nautobot/users/tests/test_api.py

```diff
@@ -73,6 +73,69 @@ def setUpTestData(cls):
         Group.objects.create(name="Group 3")
 
 
+class TokenTest(APIViewTestCases.APIViewTestCase):
+    model = Token
+    brief_fields = ["display", "id", "key", "url", "write_enabled"]
+    bulk_update_data = {
+        "description": "New description",
+    }
+
+    def setUp(self):
+        super().setUp()
+
+        tokens = (
+            # We already start with one Token, created by the test class
+            Token(user=self.user),
+            Token(user=self.user),
+        )
+        # Use save() instead of bulk_create() to ensure keys get automatically generated
+        for token in tokens:
+            token.save()
+
+        self.create_data = [
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+        ]
+
+    def test_provision_token_valid(self):
+        """
+        Test the provisioning of a new REST API token given a valid username and password.
+        """
+        data = {
+            "username": "user1",
+            "password": "abc123",
+        }
+        user = User.objects.create_user(**data)
+        url = reverse("users-api:token_provision")
+
+        response = self.client.post(url, **self.header, data=data)
+        self.assertEqual(response.status_code, 201)
+        self.assertIn("key", response.data)
+        self.assertEqual(len(response.data["key"]), 40)
+        token = Token.objects.get(user=user)
+        self.assertEqual(token.key, response.data["key"])
+
+    def test_provision_token_invalid(self):
+        """
+        Test the behavior of the token provisioning view when invalid credentials are supplied.
+        """
+        data = {
+            "username": "nonexistentuser",
+            "password": "abc123",
+        }
+        url = reverse("users-api:token_provision")
+
+        response = self.client.post(url, **self.header, data=data)
+        self.assertEqual(response.status_code, 403)
+
```

Comment:
Would you mind adding one more test case, where the `username` corresponds to a valid User but the `password` is incorrect? I don't expect any issue here but it would be good to have test coverage.

Reply:
Done.

Renamed this to ` test_provision_token_invalid_username` and added ` test_provision_token_invalid_password`